serializer.rst

```diff
@@ -2,223 +2,443 @@ How to Use the Serializer
 =========================
 
 Symfony provides a serializer to serialize/deserialize to and from objects and
-different formats (e.g. JSON or XML). Before using it, read the
-:doc:`Serializer component docs </components/serializer>` to get familiar with
-its philosophy and the normalizers and encoders terminology.
+different formats (e.g. JSON or XML).
 
 .. _activating_the_serializer:
 
 Installation
 ------------
 
 In applications using :ref:`Symfony Flex <symfony-flex>`, run this command to
-install the ``serializer`` :ref:`Symfony pack <symfony-packs>` before using it:
+install the serializer :ref:`Symfony pack <symfony-packs>` before using it:
 
 .. code-block:: terminal
 
     $ composer require symfony/serializer-pack
 
-Using the Serializer Service
-----------------------------
+.. note::
 
-Once enabled, the serializer service can be injected in any service where
-you need it or it can be used in a controller::
+    The serializer pack also installs some commonly used optional
+    dependencies of the Serializer component. When using this component
+    outside the Symfony framework, you might want to start with the
+    ``symfony/serializer`` package and install optional dependencies if you
+    need them.
 
-    // src/Controller/DefaultController.php
-    namespace App\Controller;
+.. seealso::
 
-    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
-    use Symfony\Component\Serializer\SerializerInterface;
+    A popular alternative to the Symfony Serializer component is the third-party
+    library, `JMS serializer`_ (versions before ``v1.12.0`` were released under
+    the Apache license, so incompatible with GPLv2 projects).
 
-    class DefaultController extends AbstractController
+Serializing an Object
+---------------------
+
+For this example, assume the following class exists in your project::
+
+    // src/Model/Person.php
+    namespace App\Model;
+
+    class Person
     {
-        public function index(SerializerInterface $serializer)
+        private int $age;
+        private string $name;
+        private bool $sportsperson;
+
+        public function __construct(string $name, int $age, bool $sportsperson)
+        {
+            $this->age = $age;
+            $this->name = $name;
+            $this->sportsperson = $sportsperson;
+        }
+
+        public function getAge(): int
+        {
+            return $this->age;
+        }
+
+        public function getName(): string
         {
-            // keep reading for usage examples
+            return $this->name;
+        }
+
+        public function isSportsperson(): bool
+        {
+            return $this->sportsperson;
         }
     }
 
-Or you can use the ``serialize`` Twig filter in a template:
+If you want to serialize this object into JSON, you need to get the
```

Comment:
Instead of focusing on "what" (serialize/deserialize) I'd focus at the beginning of the article on the "why" (transform some data for some reason).

For example:

**Before**:

```rst
If you want to serialize this object into JSON, you need to get the
``serializer`` service by using the
:class:`Symfony\\Component\\Serializer\\SerializerInterface` parameter type:
```

**After**:

```rst
If you want to transform objects of this type into a JSON structure to send them
via an API response, get the ``serializer`` service by using the
:class:`Symfony\\Component\\Serializer\\SerializerInterface` parameter type:
```

Reply:
Great suggestion